### PR TITLE
Use MarkStatus* helpers in pkg/pod/status

### LINKED ISSUE #2770: Make MarkResource* helpers for Pod similar to what we have for TaskRun and Pipelinerun
# Expected Behavior
Less boilerplate code in `pod/status.go`

# Actual Behavior

We repeat this code several times in small variations:
```
		trs.SetCondition(&apis.Condition{
			Type:    apis.ConditionSucceeded,
			Status:  corev1.ConditionFalse,
			Reason:  v1beta1.TaskRunReasonFailed.String(),
			Message: msg,
		})
```
We could created helpers similar to what we did for TaskRun and PipelineRun, however since we do not own the `Pod` API we could just add them to `status.go` directly.

/kind cleanup

### PR DESCRIPTION
**Pull request by adshmh**
# Changes
Use MarkStatus* helper functions in pkg/pod/status.go
Closes #2770

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

### CODE REVIEW

File: pkg/pod/status.go [hunk 1]
```diff
@@ -101,12 +101,7 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 	trs := &tr.Status
 	if trs.GetCondition(apis.ConditionSucceeded) == nil || trs.GetCondition(apis.ConditionSucceeded).Status == corev1.ConditionUnknown {
 		// If the taskRunStatus doesn't exist yet, it's because we just started running
-		trs.SetCondition(&apis.Condition{
-			Type:    apis.ConditionSucceeded,
-			Status:  corev1.ConditionUnknown,
-			Reason:  v1beta1.TaskRunReasonRunning.String(),
-			Message: "Not all Steps in the Task have finished executing",
-		})
+		MarkStatusRunning(trs, "", "")
```
Comment (vdemeester) [inline pkg/pod/status.go:104]: We need to keep the message, don't we ?

File: pkg/pod/status.go [hunk 2]
```diff
@@ -206,12 +191,7 @@ func updateCompletedTaskRun(trs *v1beta1.TaskRunStatus, pod *corev1.Pod) {
 func updateIncompleteTaskRun(trs *v1beta1.TaskRunStatus, pod *corev1.Pod) {
 	switch pod.Status.Phase {
 	case corev1.PodRunning:
-		trs.SetCondition(&apis.Condition{
-			Type:    apis.ConditionSucceeded,
-			Status:  corev1.ConditionUnknown,
-			Reason:  v1beta1.TaskRunReasonRunning.String(),
-			Message: "Not all Steps in the Task have finished executing",
-		})
+		MarkStatusRunning(trs, "", "")
```
Comment (vdemeester) [inline pkg/pod/status.go:194]: Same here

File: pkg/pod/status.go (hunk 1 above)
Comment (afrittoli) [reply pkg/pod/status.go:104]: And the reason!

File: pkg/pod/status.go (hunk 2 above)
Comment (adshmh) [reply pkg/pod/status.go]: Thank you for the review. Fixed.

File: pkg/pod/status.go (hunk 1 above)
Comment (adshmh) [reply pkg/pod/status.go]: Thank you for the review. Fixed.

**Comment by adshmh** [comment]
> Thanks for working on this!
> I agree we need to keep reason and message.
> 
> It seems worrying that all test pass with them missing, sounds like we lack some unit test coverage.
> As part of this, could you add some unit test coverage?

Thank you for the review. I reviewed the code again, the tests were passing because I was setting default values for `Reason` and `Message` fields in the helper function. I have updated the helper function's signature to fix this which was causing confusion. The tests seem to cover all cases involving helper functions.

File: pkg/pod/status.go [hunk 3]
```diff
@@ -359,6 +334,33 @@ func getWaitingMessage(pod *corev1.Pod) string {
 	return "Pending"
 }
 
+func MarkStatusRunning(trs *v1beta1.TaskRunStatus, reason, message string) {
```
Comment (bobcatfish) [inline pkg/pod/status.go:337]: as public exported functions, the convention is to include docstrings

Comment (adshmh) [reply pkg/pod/status.go:344]: Thank you for the review. Fixed.

**Comment by pritidesai** [comment]
This looks good and as an additional effort/PR, we could update many unit tests from the same test file `status_test.go` to use these helpers, e.g. running condition [here](https://github.com/tektoncd/pipeline/blob/master/pkg/pod/status_test.go#L36), failed condition [here](https://github.com/tektoncd/pipeline/blob/master/pkg/pod/status_test.go#L214),  etc instead of creating `apis.Condition` over and over.

/lgtm

**Pull request merged**